### ISSUE TITLE
Gradient anomaly after Addition

### ISSUE THREAD

**Issue by lfchener** [pytorch/pytorch#41466]
## 🐛 Bug

I ran [Tacotron2]( https://github.com/NVIDIA/tacotron2) model, and checked gradient of every temporary variables.
When i checked [here](https://github.com/NVIDIA/tacotron2/blob/master/model.py#L59), I got an unexpected result. Hope you can solve my confusion.

## To Reproduce

1. Run [Tacotron2]( https://github.com/NVIDIA/tacotron2) follow readme.
2. Modified code like this:
![image](https://user-images.githubusercontent.com/6771821/87530846-7f024900-c6c3-11ea-9ec9-707aa3c2c823.png)
 tmp*_grads are some global list variables.
3. and then add breakpoint at [here](https://github.com/NVIDIA/tacotron2/blob/master/train.py#L236) after compute grad. 
![image](https://user-images.githubusercontent.com/6771821/87530371-d522bc80-c6c2-11ea-83ea-47eea9ab31ba.png)
4. then can  print grads of tmp*_grads like:
tmp1_grads[-1].grad[20, 20, :10]

result like:
![image](https://user-images.githubusercontent.com/6771821/87544382-eaa2e100-c6d8-11ea-8d62-7dddc63409ae.png)

## Expected behavior
In theory, i will get tmp1_grads = tmp2_grads = tmp3_grads = tmp4_grads
but, i only get tmp3_grads = tmp1_grads
and i could't find any other code use tmp2 and tmp4
so, why i get this result？

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.5
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6
 - CUDA/cuDNN version: CUDA10
 - GPU models and configuration: same as README
 - Any other relevant information:

## Additional context

cc @ezyang @SsnL @albanD @zou3519 @gqchen

**Comment by albanD** [comment]
Hi,

Sorry for the late answer butwe use github issues only for bugs or feature requests.
Please use the forum to ask questions: https://discuss.pytorch.org/ as you will get a much faster answer there.

It would be nice if you could share a small code sample that we can run to reproduce this as well when you open a thread there.
